src/ui/inventory.js

```diff
@@ -61,12 +63,32 @@ export class Inventory {
     addToInventory(item, createSprite = true) {
         // Search for empty slot
         for (let index = 0; index < this.playerItems.length; index++) {
-            // Check if object exists
-            if (!Object.keys(this.playerItems[index]).length) {
+            let itemExists = Object.keys(this.playerItems[index]).length;
+
+            // Check if it can stack with other items
+            if (
+                itemExists &&
+                item.stackable &&
+                this.playerItems[index].item == item.item &&
```

Comment:
Can we simplify this by checking for the name?

Reply:
No because it's checking through the cookies which has the item/type and doesn't have the "name" only so that's why I check both. You also need to check if it exists or you'll get errors on empty inventory slots